mmcv/ops/csrc/pytorch/cuda/rans.cu

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) OpenMMLab. All rights reserved
+
+#include "pytorch_cuda_helper.hpp"
+#include "rans_cuda_kernel.cuh"
+
+// ------------------- RANS ENCODING -------------------
+std::string RansEncodeWithIndexesCUDAKernelLauncher(
+    const Tensor symbols, const Tensor indexes, const Tensor cdfs,
+    const Tensor cdfs_sizes, const Tensor offsets, int num_threads) {
+  // param symbols: (N, )
+  // param indexes: (N, )
+  // param cdfs: (M, max_cdf_bin_size)
+  // param cdfs_sizes: (M, )
+  // param offsets: (M, )
+
+  // check input
+  check_rans_encode_input(symbols, indexes, cdfs, cdfs_sizes, offsets,
+                          num_threads);
+
+  // set device
+  at::cuda::CUDAGuard device_guard(symbols.device());
+  cudaStream_t stream = at::cuda::getCurrentCUDAStream();
+
+  // count number of symbols
+  const int num_symbols = symbols.size(0);
+  const int num_cdfs = cdfs.size(0);
+  const int max_cdf_bin_size = cdfs.size(1);
+
+  // set kernel params
+  int cuda_threads = min(num_symbols, THREADS_PER_BLOCK);
+  dim3 blocks((num_symbols + cuda_threads - 1) / cuda_threads);
+  dim3 threads(cuda_threads);
+
+  // malloc device memory for cache results
+  RansSymbol *rans_symbols;  // store encoded symbols
+  CUDA_MALLOC(rans_symbols, num_symbols);
```

Comment:
Do not use cudamalloc here. you can create a int8 tensor with the size you want. reinterprate the data_ptr into the RansSymbol

Reply:
If you do so. you do not need synchronize the stream and move to cpu. just leave it to pytorch.

Reply:
I have changed it into torch tensor.
```suggestion
   auto rans_symbols_tensor =
      CREATE_TENSOR(num_symbols * sizeof(RansSymbol), torch::ScalarType::Byte,
                    symbols.device());  // store encoded symbols;
  RansSymbol *rans_symbols = (RansSymbol *)rans_symbols_tensor.data_ptr();
```